pkg/monitor/lease.go

```diff
@@ -2,71 +2,107 @@ package monitor
 
 import (
 	"fmt"
+	"io/ioutil"
 	"math"
 	"net"
 	"os"
 	"os/exec"
 	"path/filepath"
+	"regexp"
 
-	"github.com/openshift/baremetal-runtimecfg/pkg/config"
-	"github.com/openshift/baremetal-runtimecfg/pkg/utils"
 	"github.com/sirupsen/logrus"
 	"github.com/vishvananda/netlink"
+	"gopkg.in/fsnotify.v1"
+	"gopkg.in/yaml.v2"
 )
 
 const MonitorConfFileName = "unsupported-monitor.conf"
 const leaseFile = "lease-%s"
-const macAddressBytes = 6
 
-var MacPrefixQumranet = [...]byte{0x00, 0x1A, 0x4A}
-
-type VIP struct {
-	name string
-	ip   net.IP
+type vip struct {
+	Name       string `yaml:"name"`
+	MacAddress string `yaml:"mac-address"`
+	IpAddress  string `yaml:"ip-address"`
+}
+type yamlVips struct {
+	APIVip     *vip `yaml:"api-vip"`
+	IngressVip *vip `yaml:"ingress-vip"`
 }
 
-func needLease(cfgPath string) (bool, error) {
+func getVipsToLease(cfgPath string) (vips *yamlVips, err error) {
 	monitorConfPath := filepath.Join(filepath.Dir(cfgPath), MonitorConfFileName)
 
-	_, err := os.Stat(monitorConfPath)
+	info, err := os.Stat(monitorConfPath)
 
-	if err == nil {
+	if err == nil && info.Mode().IsRegular() {
 		log.WithFields(logrus.Fields{
 			"file": monitorConfPath,
 		}).Info("Monitor conf file exist")
-		return true, nil
+
+		data, err := ioutil.ReadFile(monitorConfPath)
+
+		if err != nil {
+			log.WithFields(logrus.Fields{
+				"filename": monitorConfPath,
+			}).WithError(err).Error("Failed to read monitor file")
+			return nil, err
+		}
+
+		return parseMonitorFile(data)
 	}
 	if os.IsNotExist(err) {
 		log.WithFields(logrus.Fields{
 			"file": monitorConfPath,
 		}).Info("Monitor conf file doesn't exist")
-		return false, nil
+		return nil, nil
 	}
 
 	log.WithFields(logrus.Fields{
 		"file": monitorConfPath,
 	}).WithError(err).Error("Failed to get file status")
-	return false, err
+	return nil, err
+}
+
+func parseMonitorFile(buffer []byte) (*yamlVips, error) {
+	var vips yamlVips
+
+	if err := yaml.Unmarshal(buffer, &vips); err != nil {
+		log.WithFields(logrus.Fields{
+			"buffer": buffer,
+		}).WithError(err).Error("Failed to parse monitor file")
+		return nil, err
+	}
+
+	if vips.APIVip == nil || vips.IngressVip == nil {
+		err := fmt.Errorf("APIVip or IngressVIP is missing from the yaml content")
+		log.Error(err)
+		return nil, err
+	}
+
+	log.Info("Valid file format")
+
+	return &vips, nil
 }
 
-func leaseVIPs(cfgPath string, clusterName string, vips []VIP) error {
+func LeaseVIPs(log logrus.FieldLogger, cfgPath string, clusterName string, vipMasterIface string, vips []vip, runInfinite bool) error {
```

Comment:
It can use and it actually uses the same one when it is called from `dynkeepalived.go`
```golang
if err = LeaseVIPs(log, cfgPath, vipIface.Name, []vip{*vips.APIVip, *vips.IngressVip}, true); err != nil {
```

We want to use the leasing functions from the `monitor` package in the `assisted-service` as well.
You define your own log in the monitor package via `var log = logrus.New()`.
In case another package with their own logger wants to use it (or even for it to be manipulated for the tests) it would be impossible without giving a log parameter.
Hence a log parameter is more flexible

Reply:
In the future, when adding a new use case for code like this it would be a good thing to highlight in the PR description. That's important context for reviewers.

Reply:
Personally, I would add new parameters to the end of the signature, not the beginning. Also, I'm not a huge fan of bare booleans. There is no context at the caller as to what a magic "true" or "false" might mean.